From f914e84e53bc9c25ad3f48c414024061168c0b69 Mon Sep 17 00:00:00 2001
From: Raviteja Sunkara <raviteja@codeaurora.org>
Date: Thu, 7 Feb 2013 12:09:44 +0530
Subject: [PATCH] Use the parent Change on the same branch for rebases

Since there can be multiple changes with the same
commit on different branches, use the parent change
on the same branch during rebase.

Change-Id: I156d5e73f4f5709463f3c2b5680b8e8395a5dc7f
---
 .../gerrit/server/changedetail/RebaseChange.java   |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/gerrit-server/src/main/java/com/google/gerrit/server/changedetail/RebaseChange.java b/gerrit-server/src/main/java/com/google/gerrit/server/changedetail/RebaseChange.java
index 6ae8c96..5000640 100644
--- a/gerrit-server/src/main/java/com/google/gerrit/server/changedetail/RebaseChange.java
+++ b/gerrit-server/src/main/java/com/google/gerrit/server/changedetail/RebaseChange.java
@@ -234,8 +234,7 @@ public class RebaseChange {
         depPatchSetList = db.patchSets().byRevision(ancestorRev).toList();
       }
 
-      if (!depPatchSetList.isEmpty()) {
-        PatchSet depPatchSet = depPatchSetList.get(0);
+      for (PatchSet depPatchSet : depPatchSetList) {
 
         Change.Id depChangeId = depPatchSet.getId().getParentKey();
         Change depChange;
@@ -245,6 +244,9 @@ public class RebaseChange {
         } else {
           depChange = depChangeList.get(0);
         }
+        if (!depChange.getDest().equals(destBranch)) {
+          continue;
+        }
 
         if (depChange.getStatus() == Status.ABANDONED) {
           throw new IOException("Cannot rebase a change with an abandoned parent: "
@@ -260,6 +262,7 @@ public class RebaseChange {
               db.patchSets().get(depChange.currentPatchSetId());
           baseRev = latestDepPatchSet.getRevision().get();
         }
+        break;
       }
 
       if (baseRev == null) {
-- 
1.7.8.3

